include/ftof.h

```diff
@@ -0,0 +1,28 @@
+/*! \file ftof.h
+ *  \brief Main header file for Fast/Fenics interface
+ */
+
+
+#ifndef __FTOF_H
+#define __FTOF_H
+
+#include <dolfin.h>
+
+extern "C"
+{
+#include "fasp.h"
+#include "fasp_functs.h"
+    INT fasp_solver_bdcsr_krylov_block_3(block_dCSRmat *A,
+                                       dvector *b,
+                                       dvector *x,
+                                       itsolver_param *itparam,
+                                       AMG_param *amgparam,
+                                       dCSRmat *A_diag);
+#define FASP_BSR     ON  /** use BSR format in fasp */
```

Comment:
also add "free(bsr_A.IA);" at the end of test_ftof that's the only thing
that is not free by dolfin

also I think it might be better to put all the name of the files in lower
case; FtoF -> ftof

On Sun, Oct 4, 2015 at 6:12 PM, Maximilian Metti <notifications@github.com>
wrote:

> In include/ftof.h
> <https://github.com/thepnpsolver/modular-pnp/pull/3#discussion_r41103190>:
>
> > +#ifndef __FTOF_H
> > +#define __FTOF_H
> > +
> > +#include <dolfin.h>
> > +
> > +extern "C"
> > +{
> > +#include "fasp.h"
> > +#include "fasp_functs.h"
> > +    INT fasp_solver_bdcsr_krylov_block_3(block_dCSRmat *A,
> > +                                       dvector *b,
> > +                                       dvector *x,
> > +                                       itsolver_param *itparam,
> > +                                       AMG_param *amgparam,
> > +                                       dCSRmat *A_diag);
> > +#define FASP_BSR     ON  /** use BSR format in fasp */
>
> Nice. This is a good place for this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thepnpsolver/modular-pnp/pull/3/files#r41103190>.
>